### PR TITLE
NEXUS-8758 - Index creation does not take name case into account

### PR DESCRIPTION
**Pull request by kellyrob99**
- compare index names case-insensitively when checking for existence

https://issues.sonatype.org/browse/NEXUS-8758
http://bamboo.s/browse/NX3-OSSF392-1

### CODE REVIEW

File: components/nexus-repository/src/main/java/org/sonatype/nexus/repository/search/SearchServiceImpl.java [hunk 1]
```diff
@@ -100,7 +100,8 @@ public void createIndex(final Repository repository) {
     checkNotNull(repository);
     // TODO we should calculate the checksum of index settings and compare it with a value stored in index _meta tags
     // in case that they not match (settings changed) we should drop the index, recreate it and re-index all components
-    if (!client.get().admin().indices().prepareExists(repository.getName()).execute().actionGet().isExists()) {
+    if (!client.get().admin().indices().prepareExists(Strings2.lower(repository.getName())).execute().actionGet()
```
Comment (mcculls) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/search/SearchServiceImpl.java]: Might be safer to reuse the existing `safeIndexName(repository)` method for consistency?

Comment (kellyrob99) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/search/SearchServiceImpl.java]: indeed, forgot entirely about that :(

**Comment by mcculls** [comment]
+1

**Comment by jdillon** [comment]
http://bamboo.s/browse/NX3-OSSF392-2 :white_check_mark:

**Comment by jdillon** [comment]
+1

**Comment by jdillon** [comment]
manually merged

**Pull request closed without merging**